File: recipes/jvm/java-recipes/micronaut-postgres-recipe/src/main/java/com/trendyol/stove/recipes/micronaut/domain/Product.java
package com.trendyol.stove.recipes.micronaut.domain;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.serde.annotation.Serdeable;
import java.util.Date;

@Serdeable
@MappedEntity("products")
public record Product(
    @Id String id, String name, long supplierId, boolean isBlacklist, Date createdDate) {

  public static Product create(String id, String name, long supplierId, boolean isBlacklist) {
    return new Product(id, name, supplierId, isBlacklist, new Date());
  }
}

File: examples/micronaut-example/src/main/kotlin/stove/micronaut/example/application/domain/Product.kt
package stove.micronaut.example.application.domain

import io.micronaut.serde.annotation.Serdeable
import java.util.*

@Serdeable
data class Product(
  val id: String,
  val name: String,
  val supplierId: Long,
  val isBlacklist: Boolean,
  val createdDate: Date
) {
  companion object {

    fun new(id: String, name: String, supplierId: Long, isBlacklist: Boolean): Product {
      return Product(
        id = id,
        name = name,
        supplierId = supplierId,
        createdDate = Date(),
        isBlacklist = isBlacklist
      )
    }
  }
}
